otelcollector/build/linux/Dockerfile

```diff
@@ -101,7 +101,7 @@ RUN cmake \
 RUN make
 RUN make install
 
-FROM mcr.microsoft.com/cbl-mariner/base/core:2.0 as builder
+FROM mcr.microsoft.com/azurelinux/base/core:3.0 as builder
```

Comment:
We have one more dockerfile that builds the target-allocator step at otelcollector/otel-allocator/Dockerfile, could you please update to 3.0 there too? Luckily there's no .so files that need to be copied over :)